### PR TITLE
Markvalenzuela72/ver 25

### PR DESCRIPTION
**Pull request by markvalenzuela72**

### CODE REVIEW

File: app/(auth)/signup/page.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import Link from "next/link";
+"use client";
```
Comment (krizoeris) [inline app/(auth)/signup/page.tsx]: Prefer to always use 'use server' in pages

File: app/types/definitions.ts
Comment (krizoeris) [inline app/types/definitions.ts:1]: Rename the file name specific to defined types. e.g. definitions -> users

File: app/(auth)/signup/page.tsx [hunk 2]
```diff
@@ -22,32 +29,64 @@ export default function SignUpPage() {
       </CardHeader>
       <CardContent>
         <div className="grid gap-4">
-          <div className="grid grid-cols-2 gap-4">
+          <form action={dispatch}>
```
Comment (krizoeris) [inline app/(auth)/signup/page.tsx]: Separate the form in different component. Prefer to put in './component/auth/SignupForm.tsx' or './app/(auth)/signup/signupForm.tsx'.

File: actions/userActions.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import { redirect } from "next/navigation";
+
+const FormSchema = z.object({
+  id: z.string(),
+  name: z.string({
+    invalid_type_error: "Please select a name.",
+  }),
+  email: z
+    .string({
+      invalid_type_error: "Please select a customer.",
```
Comment (krizoeris) [inline actions/userActions.ts]: markvalenzuela72@gmail.com

File: db/drizzle.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import postgres from "postgres";
+import { drizzle } from "drizzle-orm/postgres-js";
+import "dotenv/config";
```
Comment (krizoeris) [inline db/drizzle.ts]: No need to install dotenv, you can just import '../envConfig' that uses built in '@next/env'

File: actions/userActions.ts (hunk 3 above)
Comment (markvalenzuela72) [reply actions/userActions.ts]: > Change "Please select a customer." -> "Please insert an email address" Fixed

**Review by ROASAgency — APPROVED** [review]

**Review by krizoeris — APPROVED** [review]

**Pull request merged**